src/dswx_sar/filter_SAR.py

```diff
@@ -122,11 +125,210 @@ def lee_enhanced_filter(img, win_size=3, k=K_DEFAULT, cu=CU_DEFAULT,
+    :return: The label of the largest NaN region and an array where each
+             contiguous NaN region is labeled with a unique integer.
+    """
+    # Identify NaNs and invert, because ndimage.label labels non-zero regions
+    nan_mask = np.isnan(data)
+    labeled_regions, num_features = ndimage.label(~nan_mask)
+
+    # Count the occurrences of each label (excluding the background, which is 0)
+    label_sizes = np.bincount(labeled_regions.ravel())[1:]  # Exclude label 0
+
+    # Find the label of the largest NaN region
+    largest_label = np.argmax(label_sizes) + 1  # +1 because we excluded label 0 earlier
+
+    return largest_label, labeled_regions
+
+
+def interpolate_except_largest_nan_region(data):
+    """
+    Interpolates to fill NaN values for areas not covered by the largest contiguous NaN region.
+
+    :param data: 2D numpy array with NaNs.
+    :return: 2D numpy array with NaNs filled, except for the largest NaN region.
+    """
+    nan_mask = np.isnan(data)
+    largest_label, labeled_regions = find_largest_nan_polygon(data)
+
+    # Mask for the largest NaN region, which we'll exclude from filling
+    largest_nan_mask = (labeled_regions == largest_label)
+
+    # Find indices of NaNs that are not in the largest NaN region
+    fill_indices = np.where(nan_mask & ~largest_nan_mask)
+
+    # Coordinates of points for interpolation (excluding largest NaN region and non-NaN)
+    points = np.column_stack((np.where(~nan_mask & ~largest_nan_mask)))
+    values = data[~nan_mask & ~largest_nan_mask]
+
+    # Coordinates where we want to interpolate (excluding largest NaN region)
+    xi = np.column_stack(fill_indices)
+
+    if xi.size > 0 and values.size > 0:  # Ensure there are points to interpolate
+        # Perform interpolation
+        interpolated_values = griddata(points, values, xi, method='nearest')
+
+        # Insert interpolated values back into the original data array
+        data[fill_indices] = interpolated_values
+    else:
+        data = -30
+    return data
+
+
+def anisotropic_diffusion(img, **kwargs):
+    """
+    Anisotropic Diffusion
+
+    Parameters
+    ----------
+    img: numpy.ndarray
+        2 dimensional array which has real intensity values
+    weight: float
+        window size to apply filter
+
+    Returns
+    -------
+    filter_im: numpy.ndarray
+        filtered intensity image.
+    """
+    print('anisotropic_diffusion', kwargs)
+    weight = kwargs.get('weight', 1)
+    mask = np.isnan(img)
+
+    img_db = 10 * np.log10(img)
+    img_db_filled = fill_nan_value(img_db)
+    filtered_img = denoise_tv_chambolle(img_db_filled,
+                                        weight=weight,
+                                        )
+    # Vectorize conditional replacements using masks
+    filtered_img[mask] = np.nan  # Preserve original NaN positions
+    # zero_or_negative_mask = filtered_img <= 0
+    infinite_mask = np.isinf(filtered_img) | np.isinf(-filtered_img)
```

Comment:
A heads up that `|` operator might not be necessary because `np.isinf()` will return `True` regardless of the sign of `inf`. Toy code as below:
```python
a = np.array([-float('inf'), float('inf')])
np.isinf(a)  # array([ True,  True])
```